primehub/apps.py

```diff
@@ -143,37 +145,106 @@ def create(self, config: dict):
-            for x in config['env']:
-                if not isinstance(x, dict):
-                    invalid_config('entry in the "env" should be a name and value pair')
-                valid = 'name' in x and 'value' in x
-                if not valid:
-                    invalid_config('entry in the "env" should be a name and value pair')
-                if not isinstance(x['value'], str):
-                    invalid_config(f'value in an entry must be the string type => {x}')
+        self.apply_auto_filling(config)
+        self.validate_creation(config)
 
-        self._verify_dependency(config)
         results = self.request({'data': config}, query)
         if 'data' in results:
             return results['data']['createPhApplication']
         return results
 
+    @cmd(name='update', description='Update an application', optionals=[('file', file_flag)])
+    def _update_cmd(self, id: str, **kwargs):
+        """
+        Update a PrimeHub application
+
+        :type id: str
+        :param id: The id of PrimeHub application
+
+        :type file: str
+        :param file: The file path of PrimeHub application configuration
+
+        :rtype dict
+        :return The information of the PrimeHub application
+        """
+
+        config = primehub_load_config(filename=kwargs.get('file', None))
+        if not config:
+            invalid_config('PrimeHub application configuration is required.')
+        return self.update(id, config)
+
+    def update(self, id: str, config: dict):
+        """
+        Update a PrimeHub application
+
+        :type id: str
+        :param id: The id of PrimeHub application
+
+        :type config: dict
+        :param config: The file path of PrimeHub application configuration
+
+        :rtype dict
+        :return The information of the PrimeHub application
+        """
+
+        self.validate_updating(config)
+        query = """
+        mutation UpdatePhApplication(
+          $where: PhApplicationWhereUniqueInput!
+          $data: PhApplicationUpdateInput!
+        ) {
+          updatePhApplication(where: $where, data: $data) {
+            id
+          }
+        }
+        """
+
+        # config without instanceType will make the API crash
+        if 'instanceType' not in config:
+            current = self.get(id)
+            if not current:
+                return None
+            config['instanceType'] = current['instanceType']
+
+        results = self.request({'data': config, 'where': {'id': id}}, query)
+        if 'data' in results:
+            return results['data']['updatePhApplication']
+        return results
+
+    def apply_auto_filling(self, config):
+        if 'id' not in config:
+            config['id'] = auto_gen_id(config['templateId'])
+        if 'env' not in config:
+            template = self.primehub.apptemplates.get(config['templateId'])
+            config['env'] = [{'name': x['name'], 'value': x['defaultValue']} for x in template['defaultEnvs']]
```

Comment:
I found that `template['defaultEnvs']` of `code-server` is `None`, and it would cause `TypeError: 'NoneType' object is not iterable`.